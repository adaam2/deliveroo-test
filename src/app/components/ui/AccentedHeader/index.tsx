import React from 'react';

import styles from "./AccentedHeader.module.scss";
import { Container } from 'ui';

export interface AccentedHeaderProps {
  children: any;
  className?: string;
}

const AccentedHeader: React.SFC<AccentedHeaderProps> = ({ children, className }) => {
  return (
    <header className={styles.accentedHeader}>
      <Container className={className}>
        {children}
      </Container>
    </header>
  );
}

export default AccentedHeader;