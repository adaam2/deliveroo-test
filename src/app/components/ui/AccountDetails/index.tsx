import React from 'react';

import { Icon, P } from '../';
import styles from "./AccountDetails.module.scss";

export interface AccountDetailsProps {
  userName: string;
}

const AccountDetails: React.SFC<AccountDetailsProps> = ({ userName }) => {
  return (
    <div className={styles.wrapper}>
      <div className={styles.flexItem}>
        <Icon iconKey="user" />
      </div>
      <div className={styles.userName}>
        <P marginLeft={8} fontSize={16} colour="#2e3333">
          {userName}
        </P>
      </div>
    </div>
  );
}

export default AccountDetails;