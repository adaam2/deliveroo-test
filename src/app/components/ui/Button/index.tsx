import React from 'react';
import classNames from "classnames/bind";

import styles from "./Button.module.scss";

const cx = classNames.bind(styles);

export enum ButtonType {
  Primary,
  Secondary
}

export interface ButtonProps {
  type: ButtonType,
  children: any;
  onClick(e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
  ariaLabel?: string;
}

const Button: React.SFC<ButtonProps> = ({
  type = ButtonType.Primary,
  children,
  onClick,
  ariaLabel
}) => {
  const buttonClasses = cx({
    [styles.button]: true,
    [styles.primary]: type === ButtonType.Primary,
    [styles.secondary]: type === ButtonType.Secondary
  });

  return (
    <button aria-label={ariaLabel} onClick={onClick} className={buttonClasses}>
      {children}
    </button>
  );
}

export default Button;