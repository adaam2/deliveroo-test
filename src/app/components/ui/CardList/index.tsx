import React from 'react';

import styles from "./CardList.module.scss";

export interface CardListProps {
  children: any;
}

const CardList: React.SFC<CardListProps> = ({ children }) => {
  return (
    <div className={styles.list}>
      {children}
    </div>
  );
}

export default CardList;