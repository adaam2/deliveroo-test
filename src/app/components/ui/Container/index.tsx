import React from 'react';
import classNames from "classnames/bind";

import styles from "./Container.module.scss";

const cx = classNames.bind(styles);

export interface ContainerProps {
  children: any;
}

const Container: React.SFC<ContainerProps> = ({ children }) => {
  const containerClasses = () => cx({
    [styles.containerBase]: true
  });

  return (
    <div className={containerClasses()}>
      {children}
    </div>
  );
}

export default Container;