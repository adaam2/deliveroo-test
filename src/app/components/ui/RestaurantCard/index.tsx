import React from 'react';
import { Link } from "react-router-dom";

import { ImageCard, Title, P, Spacer } from 'ui';
import Restaurant from 'objects/restaurant';
import { FontWeight } from '../Typography';

const RestaurantCard: React.SFC<Restaurant> = ({ id, name, imageUrl, price, tags }) => {
  const tagList = tags.map(t => t.title).join(" • ");

  const priceLevel = () => Array.from(Array(price)).map(() => '£').join("");

  return (
    <Link to={`/restaurants/${id}`}>
      <ImageCard altText={name} imageUrl={imageUrl}>
        <Spacer height={8} />
        <Title fontWeight={FontWeight.Bold} fontSize={22} colour="#2e33333">
          {name}
        </Title>
        <Spacer height={5} />
        <P fontWeight={FontWeight.Regular} fontSize={14} colour="#828585">
          {tagList} {tagList.length > 0 ? ' • ' : null} {priceLevel()}
        </P>
      </ImageCard>
    </Link>
  );
}

export default RestaurantCard;