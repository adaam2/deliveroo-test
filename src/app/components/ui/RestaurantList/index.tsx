import React from 'react';

import { Container, CardList, Spacer, P, RestaurantCard } from "ui";
import Restaurant from 'objects/restaurant';

interface RestaurantListProps {
  restaurants: Restaurant[];
}

const RestaurantList: React.SFC<RestaurantListProps> = ({ restaurants }) => {
  return (
    <Container>
      <Spacer height={40} />
      <P fontSize={16} colour="#828585">{restaurants.length} Restaurants</P>
      <Spacer height={40} />

      <CardList>
        {restaurants.map((restaurant) => {
          return (
            <RestaurantCard key={restaurant.id} {...restaurant} />
          );
        })}
      </CardList>
    </Container>
  );
}

export default RestaurantList;