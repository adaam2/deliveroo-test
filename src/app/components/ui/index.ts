import Nav from "./Nav";
import Logo from "./Logo";
import Icon from "./Icon";
import { P, Title } from "./Typography";
import ImageCard from "./ImageCard";
import Container from "./Container";
import Spacer from "./Spacer";

export { Nav, Logo, Icon, P, Title, ImageCard, Container, Spacer };
