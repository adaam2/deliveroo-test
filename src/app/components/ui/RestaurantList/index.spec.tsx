import React from 'react';
import { mount, ReactWrapper } from "enzyme";

import { RestaurantList } from "ui";
import Neighbourhood from 'objects/neighbourhood';
import RestaurantCard from '../RestaurantCard';
import { StaticRouter as Router } from 'react-router';

let wrapper: ReactWrapper;

let neighbourhood : Neighbourhood = {
  title: "Shoreditch",
  restaurants: [
    {
      name: "One",
      imageUrl: "http://placehold.it/240",
      price: 1,
      id: "123",
      tags: []
    },
    {
      name: "Two",
      imageUrl: "http://placehold.it/240",
      price: 2,
      id: "234",
      tags: []
    }
  ]
}

function setup(
  neighbourhood: Neighbourhood
) {
  return mount(
    <Router>
      <RestaurantList neighbourhood={neighbourhood} />
    </Router>
  );
};

describe('<RestaurantList />', () => {
  beforeEach(() => {
    wrapper = setup(neighbourhood);
  });

  describe('When the neighbourhood has restaurants', () => {
    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders the expected number of restaurant cards', () => {
      const cards = wrapper.find(RestaurantCard);

      expect(cards.length).toEqual(2);
    });

    it('renders the restaurant count', () => {
      expect(wrapper.html())
        .toEqual(expect.stringContaining('2 Restaurants'))
    });
  });

  describe('When the neighbourhood has no restaurants', () => {
    beforeEach(() => {
      neighbourhood = {
        title: "No food for me :(",
        restaurants: []
      };

      wrapper = setup(neighbourhood);
    });

    it('doesnt render any cards', () => {
      const cards = wrapper.find(RestaurantCard);

      expect(cards.length).toEqual(0);
    });

    it('renders the "0 Restaurants" text', () => {
      expect(wrapper.html())
        .toEqual(expect.stringContaining("0 Restaurants"));
    });

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
