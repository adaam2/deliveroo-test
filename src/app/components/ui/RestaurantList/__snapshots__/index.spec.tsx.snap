// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantList /> When the neighbourhood has no restaurants matches the snapshot 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <RestaurantList
      neighbourhood={
        Object {
          "restaurants": Array [],
          "title": "No food for me :(",
        }
      }
    >
      <Container>
        <div
          className="undefined "
        >
          <Spacer
            height={40}
          >
            <div
              style={
                Object {
                  "height": 40,
                }
              }
            />
          </Spacer>
          <Component
            colour="#828585"
            fontSize={16}
          >
            <p
              style={
                Object {
                  "color": "#828585",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "lineHeight": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "opacity": undefined,
                }
              }
            >
              0
               Restaurants
            </p>
          </Component>
          <Spacer
            height={40}
          >
            <div
              style={
                Object {
                  "height": 40,
                }
              }
            />
          </Spacer>
          <CardList>
            <div />
          </CardList>
        </div>
      </Container>
    </RestaurantList>
  </Router>
</StaticRouter>
`;

exports[`<RestaurantList /> When the neighbourhood has restaurants matches the snapshot 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <RestaurantList
      neighbourhood={
        Object {
          "restaurants": Array [
            Object {
              "id": "123",
              "imageUrl": "http://placehold.it/240",
              "name": "One",
              "price": 1,
              "tags": Array [],
            },
            Object {
              "id": "234",
              "imageUrl": "http://placehold.it/240",
              "name": "Two",
              "price": 2,
              "tags": Array [],
            },
          ],
          "title": "Shoreditch",
        }
      }
    >
      <Container>
        <div
          className="undefined "
        >
          <Spacer
            height={40}
          >
            <div
              style={
                Object {
                  "height": 40,
                }
              }
            />
          </Spacer>
          <Component
            colour="#828585"
            fontSize={16}
          >
            <p
              style={
                Object {
                  "color": "#828585",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "lineHeight": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "opacity": undefined,
                }
              }
            >
              2
               Restaurants
            </p>
          </Component>
          <Spacer
            height={40}
          >
            <div
              style={
                Object {
                  "height": 40,
                }
              }
            />
          </Spacer>
          <CardList>
            <div>
              <RestaurantCard
                id="123"
                imageUrl="http://placehold.it/240"
                key="123"
                name="One"
                price={1}
                tags={Array []}
              >
                <Link
                  to="/restaurants/123"
                >
                  <a
                    href="/restaurants/123"
                    onClick={[Function]}
                  >
                    <ImageCard
                      altText="One"
                      imageUrl="http://placehold.it/240"
                    >
                      <div>
                        <img
                          alt="One"
                          src="http://placehold.it/240"
                        />
                        <div>
                          <Spacer
                            height={8}
                          >
                            <div
                              style={
                                Object {
                                  "height": 8,
                                }
                              }
                            />
                          </Spacer>
                          <Component
                            colour="#2e33333"
                            fontSize={22}
                            fontWeight={600}
                          >
                            <h1
                              style={
                                Object {
                                  "color": "#2e33333",
                                  "fontSize": 22,
                                  "fontWeight": 600,
                                  "lineHeight": undefined,
                                  "marginLeft": undefined,
                                  "marginRight": undefined,
                                  "opacity": undefined,
                                }
                              }
                            >
                              One
                            </h1>
                          </Component>
                          <Spacer
                            height={5}
                          >
                            <div
                              style={
                                Object {
                                  "height": 5,
                                }
                              }
                            />
                          </Spacer>
                          <Component
                            colour="#828585"
                            fontSize={14}
                            fontWeight={400}
                          >
                            <p
                              style={
                                Object {
                                  "color": "#828585",
                                  "fontSize": 14,
                                  "fontWeight": 400,
                                  "lineHeight": undefined,
                                  "marginLeft": undefined,
                                  "marginRight": undefined,
                                  "opacity": undefined,
                                }
                              }
                            >
                               
                               
                              Â£
                            </p>
                          </Component>
                        </div>
                      </div>
                    </ImageCard>
                  </a>
                </Link>
              </RestaurantCard>
              <RestaurantCard
                id="234"
                imageUrl="http://placehold.it/240"
                key="234"
                name="Two"
                price={2}
                tags={Array []}
              >
                <Link
                  to="/restaurants/234"
                >
                  <a
                    href="/restaurants/234"
                    onClick={[Function]}
                  >
                    <ImageCard
                      altText="Two"
                      imageUrl="http://placehold.it/240"
                    >
                      <div>
                        <img
                          alt="Two"
                          src="http://placehold.it/240"
                        />
                        <div>
                          <Spacer
                            height={8}
                          >
                            <div
                              style={
                                Object {
                                  "height": 8,
                                }
                              }
                            />
                          </Spacer>
                          <Component
                            colour="#2e33333"
                            fontSize={22}
                            fontWeight={600}
                          >
                            <h1
                              style={
                                Object {
                                  "color": "#2e33333",
                                  "fontSize": 22,
                                  "fontWeight": 600,
                                  "lineHeight": undefined,
                                  "marginLeft": undefined,
                                  "marginRight": undefined,
                                  "opacity": undefined,
                                }
                              }
                            >
                              Two
                            </h1>
                          </Component>
                          <Spacer
                            height={5}
                          >
                            <div
                              style={
                                Object {
                                  "height": 5,
                                }
                              }
                            />
                          </Spacer>
                          <Component
                            colour="#828585"
                            fontSize={14}
                            fontWeight={400}
                          >
                            <p
                              style={
                                Object {
                                  "color": "#828585",
                                  "fontSize": 14,
                                  "fontWeight": 400,
                                  "lineHeight": undefined,
                                  "marginLeft": undefined,
                                  "marginRight": undefined,
                                  "opacity": undefined,
                                }
                              }
                            >
                               
                               
                              Â£Â£
                            </p>
                          </Component>
                        </div>
                      </div>
                    </ImageCard>
                  </a>
                </Link>
              </RestaurantCard>
            </div>
          </CardList>
        </div>
      </Container>
    </RestaurantList>
  </Router>
</StaticRouter>
`;
